gemrb/plugins/OpenALAudio/OpenALAudio.cpp

```diff
@@ -504,12 +502,13 @@ std::pair<ALuint, ALuint> OpenALAudioDriver::loadSound(StringView ResRef, tick_t
 		alBufferData(buffers[0], format, channel1.data(), actualSamples * 2, samplerate);
 		alBufferData(buffers[1], format, channel2.data(), actualSamples * 2, samplerate);
 	} else {
-		short* memory = (short*) malloc(totalBytesPerChannel);
-		//multiply always with 2 because it is in 16 bits
-		unsigned int cnt1 = acm->read_samples(memory, numSamples) * 2;
-		//it is always reading the stuff into 16 bits
-		alBufferData(buffers[0], GetFormatEnum(channels, 16), memory, cnt1, samplerate);
-		free(memory);
+		std::vector<char> memory;
```

Comment:
Why not just use a char array? You don't use the extra functionality of a vector after all and it would remove the need for the reinterpret_cast.

Reply:
You mean `std::array`? That needs the size at compile time. And to my knowledge, var-length arrays only work when `new`'d, and then need to be deleted explicitly. I just try to keep unmanaged cases low.

Reply:
I meant a plain one, but yes, that would require manual memory management.
How about just using vectors of shorts and adjusting the size?